pyspeckit/spectrum/models/ammonia.py

```diff
@@ -179,41 +181,43 @@ def ammonia(xarr, tkin=20, tex=None, ntot=14, width=1, xoff_v=0.0,
                                        (Crot-Brot)*Jpara**2)/(kb*tkin))
         Zortho = 2*(2*Jortho+1)*np.exp(-h*(Brot*Jortho*(Jortho+1)+
                                            (Crot-Brot)*Jortho**2)/(kb*tkin))
+        Qpara = Zpara.sum()
+        Qortho = Zortho.sum()
         for linename in line_names:
             if ortho_dict[linename]:
                 orthoparafrac = fortho
                 Z = Zortho
+                Qtot = Qortho
                 count = ortho_count
                 ortho_count += 1
             else:
                 orthoparafrac = 1.0-fortho
                 Z = Zpara
+                Qtot = Qpara
                 count = para_count # need to treat partition function separately
                 para_count += 1
 
+
+            # for a complete discussion of these equations, please see
+            # https://github.com/keflavich/pyspeckit/blob/ammonia_equations/examples/AmmoniaLevelPopulation.ipynb
+            # and
+            # http://low-sky.github.io/ammoniacolumn/
+
             # short variable names for readability
             frq = freq_dict[linename]
             partition = Z[count]
             aval = aval_dict[linename]
 
-            # Friesen 2009 eqn A4 points out that the partition function actually says
-            # how many molecules are in the NH3(1-1) state, both upper *and* lower.
-            # population_upperlower = ntot * orthoparafrac * partition/(Z.sum())
-            # population_upperstate = population_upperlower / (1+np.exp(h*frq/(kb*tex)))
-            #
-            # Note Jan 1, 2015: This is accounted for in the eqn below.  The
-            # only difference is that I have used Tkin where Friesen et al 2009
-            # use Tex.  Since Tex describes which states are populated, that may
-            # be the correct one to use.
-
             # Total population of the higher energy inversion transition
-            population_upperstate = lin_ntot * orthoparafrac * partition/(Z.sum())
+            population_upperstate = lin_ntot * orthoparafrac * partition/Qtot
```

Comment:
Cool.  Sorry I've been less than useless here.  I'm off in getfunding
world.

However, when you say, calculate Z(J) for both states, do you just mean
"multiply by 2"?  Or is this latter "state" meaning ortho- and para-?

As for what to call them, I think "inversion energy states" is good.
Mentally, I set thus up as a pure quantum effect that has to do with the
inversion process not being time stationary and thus not a possible energy
state.

e.


On Wed, Mar 2, 2016 at 11:50 AM, Adam Ginsburg <notifications@github.com>
wrote:

> In pyspeckit/spectrum/models/ammonia.py
> <https://github.com/pyspeckit/pyspeckit/pull/136#discussion_r54770743>:
>
> >              # Total population of the higher energy inversion transition
> > -            population_upperstate = lin_ntot * orthoparafrac * partition/(Z.sum())
> > +            population_upperstate = lin_ntot * orthoparafrac * partition/Qtot
>
> I still have to reformulate the equations... which means re-re-reading my
> own notebook and making sure I put the right stuff in the right place.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyspeckit/pyspeckit/pull/136/files#r54770743>.
>



-- 
Erik Rosolowsky
Asst. Professor • Dept. of Physics
University of Alberta • +1-780-492-9272